### PR TITLE
✨ Add fields to Partcipant Entity

### PR DESCRIPTION
**Pull request by parimalak**
Added family_id, is_proband, consent_type to participant entity
Updated participant model, resource tests
Updated Mock data generation and migrations

### CODE REVIEW

File: dataservice/api/participant/models.py [hunk 1]
```diff
@@ -13,12 +13,18 @@ class Participant(db.Model, Base):
 
     :param kf_id: Unique id given by the Kid's First DCC
     :param external_id: Name given to participant by contributor
+    :param family_id: Id for the participants grouped by family
+    :param is_proband: Is participant is proband of study
```
Comment (znatty22) [inline dataservice/api/participant/models.py]: Maybe reword to "Denotes whether participant is proband of study"

File: dataservice/api/participant/models.py [hunk 2]
```diff
@@ -13,12 +13,18 @@ class Participant(db.Model, Base):
...
+    :param consent_type: Type of the consent participant belongs to
```
Comment (znatty22) [inline dataservice/api/participant/models.py]: Maybe reword to "Follows standard categorizations from dbGaP"

File: migrations/versions/f62d630e7bd7_.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+"""
```
Comment (znatty22) [inline migrations/versions/f62d630e7bd7_.py]: This might need to change if we merge Study model first

File: tests/participant/test_resources.py [hunk 4]
```diff
@@ -106,6 +112,7 @@ def test_put_participant(self):
         participant = resp['results']
         self.assertEqual(p.kf_id, participant['kf_id'])
```
Comment (znatty22) [inline tests/participant/test_resources.py:35]: This is probably my fault from before, but these asserts need to change to:
```python
self.assertEqual(p.kf_id, participant['kf_id'])
self.assertEqual(p.external_id, body['external_id'])		          
self.assertEqual(p.consent_type, body['consent_type'])
```
The reason why is bc this test should not actually be passing, since the participant resource put method hasn't been updated with the new fields. Its only passing because the participant.<new field> and p.<new field> are probably both None.

So change these assert statements and then also make sure to update the participant resource put method. We should make that deserialization dynamic in the near future so that we don't have to manually deserialize fields into the put methods of resources.

File: migrations/versions/f62d630e7bd7_.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# revision identifiers, used by Alembic.
+revision = 'f62d630e7bd7'
+down_revision = 'c1b8fdd17637'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_unique_constraint(None, 'outcome', ['kf_id'])
```
Comment (dankolbman) [inline migrations/versions/f62d630e7bd7_.py]: I don't think this belongs here?

**Review by dankolbman — COMMENTED** [review]
Probably ok for now, but we'll likely want to pull out families and consents to their own tables.

File: dataservice/api/participant/models.py [hunk 6]
```diff
@@ -13,12 +13,18 @@ class Participant(db.Model, Base):
...
     :param created_at: Time of object creation
     :param modified_at: Last time of object modification
     """
     __tablename__ = "participant"
 
     external_id = db.Column(db.Text(), doc='ID used by external study')
+    family_id = db.Column(db.Text())
+    is_proband = db.Column(db.Boolean())
```
Comment (dankolbman) [inline dataservice/api/participant/models.py]: Hmm, I wonder if this should have a default, or at least be non-nullable?

Comment (znatty22) [reply dataservice/api/participant/models.py]: Yea I think it should have a default. When I had it on the StudyParticipant table, I had set the default to false because I assumed there would be more non-probands then probands? Idk if that's the correct assumption though. @allisonheath thoughts?

Comment (allisonheath) [reply dataservice/api/participant/models.py]: I think non-nullable makes sense. I would be a bit wary about setting a default - there are downstream consequences to it being set wrong as there's several search use cases that focus specifically on probands and their families. But if you really want a default, I think there's slightly less impact if it's set to false when it's supposed to be true versus the other way around

**Review by znatty22 — APPROVED** [review]

**Review by dankolbman — APPROVED** [review]
🎉
